salt/utils/virt.py

```diff
@@ -6,16 +6,55 @@
 
 # Import python libs
 import os
+import re
 import time
 import logging
 
+from urllib.request import urlopen
+from urllib.parse import urlparse
+
 # Import salt libs
 import salt.utils.files
 
 
 log = logging.getLogger(__name__)
 
 
+def download_remote(url, dir):
+    """
+    Attempts to download a file specified by 'url'
+
+    :param url: The full remote path of the file which should be downloaded.
+    :param dir: The path the file should be downloaded to.
+    """
+
+    try:
+        remote_filename = urlparse(url).path.split('/')[-1]
+        with open(dir + '/' + remote_filename, 'wb') as file, \
+                urlopen(url) as response:
+            file.write(response.read())
+
+        return dir + '/' + remote_filename
+
+    except Exception as err:
+        raise err
+
+
+def check_remote(cmdline_path):
+    """
+    Checks to see if the path provided contains ftp, http, or https. Returns
+    the full path if it is found.
+
+    :param cmdline_path: The path to the initrd image or the kernel
+    """
+    regex = re.compile('\\b(ht|f)tps?\\b')
+
+    if regex.match(urlparse(cmdline_path.scheme)):
+        return True
+
+    return False
+
+
```

Comment:
I would combine those two in one function that downloads if remote and returns the local path.